Packs/ApiModules/Scripts/CoreIRApiModule/CoreIRApiModule.py

```diff
@@ -2942,7 +2942,43 @@ def get_script_code_command(client: CoreClient, args: Dict[str, str]) -> Tuple[s
     )
 
 
-def run_script_command(client: CoreClient, args: Dict) -> CommandResults:
+@polling_function(
+    name=demisto.command(),
+    interval=arg_to_number(demisto.args().get('polling_interval_in_seconds', 10)),
+    timeout=arg_to_number(demisto.args().get('polling_timeout', 600)),
+    requires_polling_arg=False  # means it will always be default to poll, poll=true
+)
+def script_run_polling_command(args: dict, client: CoreClient) -> PollResult:
+
+    if action_id := args.get('action_id'):
+        response = client.get_script_execution_status(action_id)
+        general_status = response.get('reply', {}).get('general_status') or ''
+
+        return PollResult(
+            response=get_script_execution_results_command(client, {'action_id': action_id}),
+            continue_to_poll=general_status.upper() in ('PENDING', 'IN_PROGRESS')
+        )
+
+    else:
+        endpoint_ids = argToList(args.get('endpoint_ids'))
+        response = get_run_script_execution_response(client, args)
+        reply = response.get('reply')
+        action_id = reply.get('action_id')
+
+        args['action_id'] = action_id
+
+        return PollResult(
+            response='',
```

Comment:
```suggestion
            response='',
```

@JudahSchwartz maybe we need to consider giving response a default value of `None` in case its not required, that is a good example.

Reply:
I think thats usually the rare case... most of the time polling is an argument. Here too, if it would be an option controlled from the outside we would need to provide something here

Reply:
I would want to normally force the programmer to provide, since the case of it getting forgotten since they dont know its required seems like it could be common

Reply:
Sounds legit, thanks for explanation